### PR TITLE
Mandrill Integration

### PR DESCRIPTION
**Pull request by asamolion**

### CODE REVIEW

File: common/djangoapps/student/forms.py [hunk 1]
```diff
@@ -80,8 +82,14 @@ def save(
...
             subject = subject.replace('\n', '')
             email = loader.render_to_string(email_template_name, context)
-            password_reset_link = loader.render_to_string('emails/password_reset_email_link.txt', context)
-            MandrillClient().send_password_reset_email(user.email, {
+
+            password_reset_link = context['protocol'] +\
+                '://' +\
+                site_name +\
+                reverse('student.views.password_reset_confirm_wrapper',
+                        kwargs={ 'uidb36': int_to_base36(user.id), 'token': context['token'] })
```
Comment (muhammadnadeem) [inline common/djangoapps/student/forms.py]: Please read about the multiple ways of string concatenation in python and which is batter. In my opinion you should use "{protocol}://{site_name}{path}".format(protocol=context['protocol'], site_name=site_name, path=path)

File: common/lib/mandrill_client/client.py [hunk 2]
```diff
@@ -6,10 +6,33 @@
...
+        arguments:
+        template_name: the slug/identifier of the mandrill email template
+        user_email: the email of the receiver
+        context: the data which is passed to the template. must be a dict
+        """
+        global_merge_vars = []
+        for key in context:
+            global_merge_vars.append(
+                { 'name': key, 'content': context[key] }
+            )
```
Comment (muhammadnadeem) [inline common/lib/mandrill_client/client.py]: you can use list comprehension here

File: lms/djangoapps/onboarding/email_utils.py [hunk 3]
```diff
@@ -23,13 +24,18 @@ def send_admin_activation_email(first_name, org_id, org_name, dest_addr, hash_ke
...
         "referring_user": hash_key.suggested_by.username,
     }
-    admin_activation_link = render_to_string('emails/admin_activation_link.txt', message_context)
 
-    message_context["key"] = admin_activation_link
+    site = safe_get_host(request)
+    if request.is_secure():
+        admin_activation_link = 'https://' + site + '/onboarding/admin_activate/' + encoded_org_id + '/' + hash_key.activation_hash
+    else:
+        admin_activation_link = 'http://' + site + '/onboarding/admin_activate/' + encoded_org_id + '/' + hash_key.activation_hash
```
Comment (muhammadnadeem) [inline lms/djangoapps/onboarding/email_utils.py]: update string concatenation logic

File: common/lib/mandrill_client/client.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        global_merge_vars = [
+            {'name': 'full_name', 'content': context['full_name']},
+            {'name': 'course_name', 'content': context['course_name']},
+            {'name': 'course_url', 'content': context['course_link']},
+        ]
+        if template_name == 'weekly':
+            global_merge_vars.append(
+                {'name': 'course_week', 'content': context['course_week']},
+            )
+        self.send_template(template_name, user_email, global_merge_vars)
```
[the remark this answers is not in the document]
Comment (asamolion) [reply common/lib/mandrill_client/client.py]: doing.

File: common/djangoapps/student/forms.py (hunk 1 above)
Comment (asamolion) [reply common/djangoapps/student/forms.py]: done.

File: common/lib/mandrill_client/client.py (hunk 2 above)
Comment (asamolion) [reply common/lib/mandrill_client/client.py]: done.

File: lms/djangoapps/onboarding/email_utils.py (hunk 3 above)
Comment (asamolion) [reply lms/djangoapps/onboarding/email_utils.py]: done.

**Comment by asamolion** [comment]
I hope this works.

**Pull request merged**